cred-manager/prisma/schema.prisma

```diff
@@ -73,4 +73,15 @@ model UserScore {
   user_type                   String                  @db.VarChar(255)
   score                       String                  @db.VarChar(255)
   created_at                  DateTime                @default(now()) @db.Timestamp(6)
+  UserScoreInterval           UserScoreInterval[]
+}
+
+model UserScoreInterval {
+  id             String    @id @default(uuid())
+  user_score_id  String
+  UserScore      UserScore @relation(fields: [user_score_id], references: [id], onDelete: Cascade)
+  score          String    @db.VarChar(255)
+  interval_start String    @db.VarChar(255)
+  interval_end   String    @db.VarChar(255)
+  created_at     DateTime  @default(now()) @db.Timestamp(6)
```

Comment:
Im afraid this could be quite costly, did you checked the interval array from one of the instances we generate? It's pretty huge, I think it could be quite unmanageable to have each user score referencing to other 5000 database objects for example. Did you consider just using an actual array of numbers instead of this table? 

Trying to consider the pros and cons, I think the only advantage here going with this approach would be that each interval would have the start and end, but afaik we could infer that from the array size and the interval start (I haven't looked too deep into this, happy to be challenged if you think this is a better approach)